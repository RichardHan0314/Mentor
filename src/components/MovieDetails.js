import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Logo from './Logo';
import SearchBtn from './SearchBtn';

const MovieDetails = ({ movieData }) => {

    return (
        <Wrapper>
            <RowHeader>
                <Logo/>
                <Link to='/'>
                    <SearchBtn />
                </Link>
            </RowHeader>
            <Row>
                <Poster image={movieData.poster_path} />
                <MovieInfo>
                    <MovieInfoCol>
                        <Title>{movieData.title}</Title>
                        <Rating>{movieData.vote_average}</Rating>
                    </MovieInfoCol>
                    <Genre>{movieData.genres.join(' ')}</Genre>
                    <MovieInfoCol>
                        <ReleaseDate>{movieData.release_date}</ReleaseDate>
                        <Runtime>{movieData.runtime} min</Runtime>
                    </MovieInfoCol>
                    <Overview>{movieData.overview}</Overview>
                </MovieInfo>
            </Row>
        </Wrapper>
    );
};

const Wrapper = styled.section`
    display: flex;
    flex-direction: column;
    width: 100%;
    box-sizing: border-box;
    padding: 30px;
    z-index: 5;
    background: #232323;
    transition: all 0.8s ease;
`;

const RowHeader = styled.div`
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 0 10px 20px 10px;
`;

const Row = styled.div`
    display: flex;
    padding: 10px;
`;

const Poster = styled.div`
    display: flex;
    justify-content: flex-end;
    width: 30%;
    height: 340px;
    background-image: url(${props => props.image});
    background-repeat: no-repeat;
    background-size: contain;
`;

const MovieInfo = styled.div`
    display: flex;
    flex-direction: column;
    color: #fff;
    padding: 16px 100px 16px 40px;
`;

const MovieInfoCol = styled.div`
    display: flex;
    align-items: center;
`;

const Title = styled.h2`
    margin: 0;
    padding-right: 20px;
    font-size: 3.5rem;
    font-weight: 400;
    letter-spacing: 3px;
`;

const Rating = styled.p`
    margin: 0;
    padding: 14px;
    color: green;
    font-size: 1.5rem;
    border: 1px solid white;
    border-radius: 50%;
`;

const Genre = styled.p`
    margin: 0;
    padding: 10px 0;
`;

const ReleaseDate = styled.p`
    margin-right: 20px;
    color: #f65261;
    font-size: 1.6rem;
`;

const Runtime = styled.p`
    margin-right: 20px;
    color: #f65261;
    font-size: 1.6rem;
`;

const Overview = styled.p`
    font-size: 1.4rem;
    letter-spacing: 1.5px;
`;

MovieDetails.propTypes = {
    // eslint-disable-next-line react/forbid-prop-types
    movieData: PropTypes.object,
};

MovieDetails.defaultProps = {
    movieData: {
        image: 'https://cdn3.vectorstock.com/i/1000x1000/50/07/http-404-not-found-error-message-hypertext-vector-20025007.jpg',
    }
};

export default MovieDetails;
