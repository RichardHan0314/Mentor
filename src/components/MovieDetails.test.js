import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import MovieDetails from './MovieDetails';

describe('MovieDetails component', () => {

    const moviesDataMock = {
        id: 1,
        poster_path: 'https://images-na.ssl-images-amazon.com/images/I/71niXI3lxlL._AC_SY679_.jpg',
        title: 'Avengers',
        genres: ['Action', 'Cartoon'],
        release_date: '2016',
        url: '',
        overview: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam accumsan commodo ligula, non gravida arcu imperdiet sed. Donec eu nibh id lectus molestie volutpat non vel sapien. Proin nec imperdiet lectus.',
        runtime: '146',
        vote_average: '4.8'
    };

    it('Render MovieDetails snapshot', () => {
        const component = render(
            <MemoryRouter>
                <MovieDetails movieData={moviesDataMock} />
            </MemoryRouter>
        );
        expect(component).toMatchSnapshot();
    });
});
