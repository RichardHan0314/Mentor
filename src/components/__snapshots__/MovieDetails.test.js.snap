// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails component Render MovieDetails snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="sc-AxgMl dQJATG"
      >
        <div
          class="sc-AxheI bTKbqT"
        >
          <p
            class="sc-AxjAm ekgtgp"
          >
            <span>
              Netflix
            </span>
            roulette
          </p>
          <a
            href="/"
          >
            <div
              class="sc-AxirZ lcfolM"
            >
              <div
                class="sc-AxiKw kmATJV"
              >
                <div
                  class="sc-AxhCb hmzFFW"
                />
                <div
                  class="sc-AxhUy byZPKr"
                />
              </div>
            </div>
          </a>
        </div>
        <div
          class="sc-Axmtr gNrMTb"
        >
          <div
            class="sc-AxmLO dsHgUp"
          />
          <div
            class="sc-fzozJi ehFhqa"
          >
            <div
              class="sc-fzoLsD gqorDw"
            >
              <h2
                class="sc-fzpans dPicaR"
              >
                Avengers
              </h2>
              <p
                class="sc-fzplWN hgmNms"
              >
                4.8
              </p>
            </div>
            <p
              class="sc-fznyAO jYVJnT"
            >
              Action Cartoon
            </p>
            <div
              class="sc-fzoLsD gqorDw"
            >
              <p
                class="sc-fznKkj NtPJI"
              >
                2016
              </p>
              <p
                class="sc-fznZeY fLCPSI"
              >
                146
                 min
              </p>
            </div>
            <p
              class="sc-fzokOt cBzZUZ"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam accumsan commodo ligula, non gravida arcu imperdiet sed. Donec eu nibh id lectus molestie volutpat non vel sapien. Proin nec imperdiet lectus.
            </p>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="sc-AxgMl dQJATG"
    >
      <div
        class="sc-AxheI bTKbqT"
      >
        <p
          class="sc-AxjAm ekgtgp"
        >
          <span>
            Netflix
          </span>
          roulette
        </p>
        <a
          href="/"
        >
          <div
            class="sc-AxirZ lcfolM"
          >
            <div
              class="sc-AxiKw kmATJV"
            >
              <div
                class="sc-AxhCb hmzFFW"
              />
              <div
                class="sc-AxhUy byZPKr"
              />
            </div>
          </div>
        </a>
      </div>
      <div
        class="sc-Axmtr gNrMTb"
      >
        <div
          class="sc-AxmLO dsHgUp"
        />
        <div
          class="sc-fzozJi ehFhqa"
        >
          <div
            class="sc-fzoLsD gqorDw"
          >
            <h2
              class="sc-fzpans dPicaR"
            >
              Avengers
            </h2>
            <p
              class="sc-fzplWN hgmNms"
            >
              4.8
            </p>
          </div>
          <p
            class="sc-fznyAO jYVJnT"
          >
            Action Cartoon
          </p>
          <div
            class="sc-fzoLsD gqorDw"
          >
            <p
              class="sc-fznKkj NtPJI"
            >
              2016
            </p>
            <p
              class="sc-fznZeY fLCPSI"
            >
              146
               min
            </p>
          </div>
          <p
            class="sc-fzokOt cBzZUZ"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam accumsan commodo ligula, non gravida arcu imperdiet sed. Donec eu nibh id lectus molestie volutpat non vel sapien. Proin nec imperdiet lectus.
          </p>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
